SU-HSDO/suhumsci
# FourKitchens Release: Sprint 16

# READY FOR REVIEW

## Summary
- SHS-131: MegaMenu Refinement: remove Level 2 items hover state if not links
- SHS-132: MegaMenu Refinement: Mobile Behavior
- SHS-138: FE: Update and redo `hb-input-focus` mixin
- SHS-95: Video media embeds in WYSIWYG collapse when aligned right
- SHS-94: Right- or left-aligned image (float) in an accordion collapses the accordion description area

## Need Review By (Date)
- 1/10

## Urgency
low

## Steps to Test
Testing SHS-131:
1. On a site with the new megamenu enabled, edit a level two menu item that has children in Drupal. Since a link is required in the link field, add <nolink> save.
2. View the new megamenu and ensure that the L2 no link item: doesn't have a hover state and has the same font-weight as the other L2s

**Testing SHS-132**:
1. On a site with the new megamenu, go to a nested child page. I liked this one /events/colloquium-series.
3. In mobile view, open the menu and see that menu item it's located under is expanded and the level 2 item is highlighted (screenshot below)
4. Refresh the page in desktop view. Click on the parent page item and see that the desktop styles remain unaffected.
![208755290-bc55b4d8-3948-4fb8-b8c0-673039985c45](https://user-images.githubusercontent.com/369018/211107098-fa4df3fd-248d-43a2-96a8-211dafc807df.png)

**Testing SHS-138**:
1. Spin up hs-colorful and go to `/default-views/courses` or view another form that utilizes the Chosen library as a select dropdown. Use `tab` to focus on the input and verify the focus styles match other form input/link focus styles
![210804696-6fb99b48-0dd3-45b0-97e0-9cd56a54bc01](https://user-images.githubusercontent.com/369018/211106849-840568d4-cd7a-45fd-aeff-03870a8260ba.png)

**Testing SHS-94:**
1. Go to the Colorful demo site. Visit the accordion component page: /components/accordions.
5. Click on the one near the top with 'Accordion with Image and Text'. When open, the image should be within the container (like the screenshot below).
6. Do the same for the Traditional site and make sure the other accordions behave as normal
![210578029-50f08c6b-2137-47fb-8553-a427019c93bd](https://user-images.githubusercontent.com/369018/211106941-1725eb89-57d1-4563-ab7b-84e4a077fbc5.png)

**Testing SHS-95**:
1. Create a flexible page and and add a text area component. Embed a YouTube video.
2. Hover over the video and click edit media. Switch between left and right and ensure the video still displays in the WYSIWYG (screenshot below)
3. Settle on 'right' aligned. Save the page and view it. Ensure that you can see the video and play it.


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
